models/class.metrilo-deleted-product-data.php

```diff
@@ -0,0 +1,97 @@
+<?php
+class Metrilo_Deleted_Product_Data
+{
+    private $db_connection;
+    
+    public function __construct()
+    {
+        global $wpdb;
+        
+        $this->db_connection = $wpdb;
+    }
+    
+    public function get_deleted_products()
+    {
+        $deleted_products        = [];
+        $deleted_product_options = [];
+        $parent_id = '';
+        $child_id  = '';
+        $qty       = '';
+        $subtotal  = '';
+        
+        $order_items_table = $this->db_connection->prefix . 'woocommerce_order_items';
+        $order_items = $this->db_connection->get_results(
+                "
+                SELECT order_items.order_item_id, order_items.order_item_name
+                FROM {$order_items_table} AS order_items
+                JOIN {$this->db_connection->order_itemmeta} AS item_meta
+                ON order_items.order_item_id = item_meta.order_item_id
+                AND item_meta.meta_key IN ('_product_id', '_variation_id')
+                WHERE item_meta.meta_value
+                NOT IN (
+                  SELECT id FROM {$this->db_connection->posts} AS posts
+                  WHERE posts.post_type IN ('product', 'product_variation')
+                )
+                AND item_meta.meta_value != 0
+                GROUP BY item_meta.meta_value
+                "
+            );
+        
+        foreach ($order_items as $item) {
+            
+            $deleted_item_meta = $this->db_connection->get_results(
+                "
+                SELECT *
```

Comment:
Why `*`? I see that you are using only the `meta_key` and the `meta_value`.

Reply:
Refactored the whole query.